content/docs/reference/buildpack-api.md

```diff
@@ -0,0 +1,147 @@
++++
+title="Buildpack API"
+weight=2
+creatordisplayname = "Joe Kutner"
+creatoremail = "jpkutner@gmail.com"
+lastmodifierdisplayname = "Joe Kutner"
+lastmodifieremail = "jpkutner@gmail.com"
++++
+
+This specification defines the interface between a buildpack and the environment that runs it.
+This API will be used by buildpack authors.
+
+A buildpack must contain three files:
+
+* `buildpack.toml`
+* `bin/detect`
+* `bin/build`
+
+The two files in `bin/` must be executable. They can be shell scripts written in a language like Bash or they
+can be executables compiled from a language like Go.
+
+## `bin/detect`
+
+### Usage
+
+```
+bin/detect PLATFORM_DIR BUILD_PLAN
+```
+
+### Summary
+
+This entrypoint is used to determine if a buildpack should
+run against a given codebase. It will often check for the existence of a particular
+file or some configuration indicating what kind of application has been provided.
+It accepts two positional arguments:
+
+* `PLATFORM_DIR` - a directory containing platform provided configuration, such as environment variables.
+* `BUILD_PLAN` - a string containing the [Build Plan](https://github.com/buildpack/spec/blob/master/buildpack.md#buildpack-plan-toml).
+
+In addition, the working directory is defined as the location of the codebase
+the buildpack will execute against.
+
+The executable must return an exit code of `0` if the codebase can be serviced by this buildpack. If the exit code is `0`, the script must print a human-readable name to `stdout`.
+
+### Example
+
+This is a simple example of a buildpack that detects a Python application by
+checking for the presence of a `requirements.txt` file:
+
+```
+#!/bin/sh
+
+if [ -f requirements.txt ]; then
+  echo "Python"
```

Comment:
Maybe say "Python Buildpack" here to make clearer that it's the buildpack name you're printing and not the language?